examples/ecs/fallible_systems.rs

```diff
@@ -77,3 +97,14 @@ fn setup(
     // Indicate the system completed successfully:
     Ok(())
 }
+
+#[derive(Resource)]
+struct UninitializedResource;
+
+fn failing_system(world: &mut World) -> Result {
+    world
+        .get_resource::<UninitializedResource>()
+        .ok_or("Resource not initialized")?;
```

Comment:
A comment here about why putting a string in here works would be helpful.

Reply:
Rust provides `impl<'a> From<&str> for Box<dyn Error + 'a>`.

See: https://doc.rust-lang.org/1.84.1/src/alloc/boxed/convert.rs.html#686

Does it make sense to add this as a comment?

Reply:
Yep! That's exactly the sort of information that's really helpful to a beginner but really hard to piece together.

Reply:
Resolved in [95b54ba](https://github.com/bevyengine/bevy/pull/17753/commits/95b54bac8e63257562d3e5d6ef805455233e17bc).